sandbox-web/src/main/java/com/exadel/sandbox/team5/controller/DiscountRestController.java

```diff
@@ -49,5 +56,19 @@ public void delete(@PathVariable Long id) {
     public List<Review> getReviewsByDiscount(@PathVariable Long discountId) {
         return reviewService.getReviewsByDiscount(discountId);
     }
+
+    @ApiOperation("Generating QR code with text \"Exadel employee. Special discount\"")
+    @PostMapping(value = "/qrcode", produces = MediaType.IMAGE_PNG_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    public ResponseEntity<byte[]> generateQRCode() {
+        try (var baos = new ByteArrayOutputStream()) {
+            var image = QRCodeGenerator.generateQRCodeImage("Exadel employee. Special discount");
+            ImageIO.write(image, "png", baos);
+            return ResponseEntity.ok(baos.toByteArray());
+        } catch (Exception e) {
+            log.error("There was an error during barcode generation", e);
+            return ResponseEntity.badRequest().build();
```

Comment:
it should be usual 500 error instead of 400. So I think it will be better to convert checked exception to unchecked in the service method

Reply:
I'm not sure that I converted it correctly, I made a new commit, pls check.